### PR TITLE
Update Travis config.

### PR DESCRIPTION
**Pull request by XhmikosR**
- remove Ruby 1.9.3 since Jekyll 3.x doesn't work on it
- specify `sudo: false` and more ruby versions
- cache gems

### CODE REVIEW

**Comment by XhmikosR** [comment]
Ok, it seems the dev gems don't get installed...

**Comment by XhmikosR** [comment]
@parkr: With my limited Ruby(gems) knowledge I can't make this work. Please check the changes and if you have any idea how to make this work let me know.

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
+source "https://rubygems.org"
 
-# Specify your gem's dependencies in octopress-paginate.gemspec
-gemspec
+gemspec :name => "octopress-paginate"
+
+gem "rake", "~> 11.0"
+
+group :development do
+  gem "clash"
```
Comment (parkr) [inline Gemfile:8]: Are these being installed? The error I see tells me that this isn't being installed...

Comment (XhmikosR) [reply Gemfile:8]: They are installed but it's like clash has no executable...

See https://travis-ci.org/octopress/paginate/jobs/155530624#L187

Comment (XhmikosR) [inline Gemfile]: See https://travis-ci.org/octopress/paginate/jobs/155530624#L187

**Pull request merged**